### ISSUE TITLE
Problems in previous RMarkdown code 

### ISSUE THREAD

**Issue by abbrenner-CBHC** [rin3-spring-2021/ED-Enhancement#5]
Hi Charlie and David,

I had my completed report and corresponding RMarkdown file, and was adapting a bit for different programs and using parameters. In the process something strange started happening and I've re-done everything, re-started, reviewed all my code, and I can't figure out why the error is occurring. Previously I used some summarize + pull code (code chunk starts on line 79 of EDMonthlyReportCode.Rmd) to generate counts and averages, and then to save the value for inline R code. Now, instead of the total = n() generating a single value of the total observations, or ave_monthly_visits = ave(visit_count) generating a single value for the average, I'm getting a vector. 

Nothing I've searched is helping me figure out what happened. In the process of creating the second report I found a small error in the way I defined something and made that change, but nothing else changed (at that point before I went on a wild goose chase), so I have no clue what made the difference.  I know the class is over, but I'm hoping you may be able to provide me with some help.  

One note - it seems that it is giving me my totals and means grouped by my client id. I didn't ask for this and I even tried running ungroup() and re-running the summarize function, but it still is grouping the data.
Allison

**Comment by charliejhadley** [comment]
Hey @abbrenner-CBHC I wasn't able to duplicate your error https://share.getcloudapp.com/lluN2vx0 could you take a look at the video and let me know what steps I need to follow to replicate the error

**Comment by charliejhadley** [comment]
I took a few more minutes to play around with adding bars for the zero counts and the only solution I had was to add another geom_col layer

```r
ggplot(data = data_ethnicity_subset,
       aes(y = motsethnicity,
           x = pct_high_freq,
           fill = motsethnicity)) +
  geom_col(aes(x = 0.25, y = motsethnicity),
           show.legend = FALSE) +
  geom_col(show.legend = FALSE) +
  scale_color_manual(
    values = c(
      "#006699",
      "#669933",
      "#0099CD",
      "#333333",
      "#999999",
      "#FF9933",
      "#006699"
    ),
    aesthetics = c("fill")
  ) +
  scale_x_continuous(labels = scales::percent_format(scale = 1)) +
  labs(title = "Clients who identify as Hispanic, but with origin \n  not specified have the highest percentage\n  of 2 or more past month ED visits",
       caption = "Figure 7. ED visit averages per client in May, by ethnicity.\n  Data include clients with 1 or more ED visits in May") +
  geom_text(
    aes(label = round(pct_high_freq, 1),
        x = ifelse(pct_high_freq == 0, 0.4, pct_high_freq - 0.5),
        color = ifelse(pct_high_freq == 0, "black", "white")),
    # hjust = 1.05,
    size = 3.5,
    fontface = "bold"
  ) +
  scale_color_identity()
```

![image](https://user-images.githubusercontent.com/9363263/121922899-497e3980-cd32-11eb-8c95-f70d4e4f918c.png)

**Comment by abbrenner-CBHC** [comment]
Hi Charlie,
I'm sorry about all the confusion around the error. I have absolutely no explanation. I tried for hours and kept running that code and ending up with vectors vs a scalar for all of those summarized values. Now it's working just fine. I closed and restarted R multiple times, so I have no idea, but it is now working as it was previously.

In terms of the bar graphs, thank you so much, this is amazing and much better than having no bars in those columns. I messed with it a bit to get the zero bars even smaller, but still present, and it's wonderful. Thank you for taking the time to help with that - I absolutely never would have even considered that as a possibility!

Allison
